cmd/validator/validator.go

```diff
@@ -72,101 +84,160 @@ func validatorUsage() {
 // will return with exit = 1
 func getFlags() (validatorConfig, error) {
 	flag.Usage = validatorUsage
+	reporterConfigFlags := reporterFlags{}
+
 	depthPtr := flag.Int("depth", 0, "Depth of recursion for the provided search paths. Set depth to 0 to disable recursive path traversal")
 	excludeDirsPtr := flag.String("exclude-dirs", "", "Subdirectories to exclude when searching for configuration files")
 	excludeFileTypesPtr := flag.String("exclude-file-types", "", "A comma separated list of file types to ignore")
-	outputPtr := flag.String("output", "", "Destination to a file to output results")
-	reportTypePtr := flag.String("reporter", "standard", "Format of the printed report. Options are standard, json, junit and sarif")
 	versionPtr := flag.Bool("version", false, "Version prints the release version of validator")
 	groupOutputPtr := flag.String("groupby", "", "Group output by filetype, directory, pass-fail. Supported for Standard and JSON reports")
 	quietPtr := flag.Bool("quiet", false, "If quiet flag is set. It doesn't print any output to stdout.")
+	flag.Var(
+		&reporterConfigFlags,
+		"reporter",
+		`A string representing report format and optional output file path separated by colon if present.
+Usage: --reporter <format>:<optional_file_path>
+Multiple reporters can be specified: --reporter json:file_path.json --reporter junit:another_file_path.xml
+Omit the file path to output to stdout: --reporter json or explicitly specify stdout using "-": --reporter json:-
+Supported formats: standard, json, junit (default: "standard")`,
+	)
 
-	flagsEnvMap := map[string]string{
-		"depth":              "CFV_DEPTH",
-		"exclude-dirs":       "CFV_EXCLUDE_DIRS",
-		"exclude-file-types": "CFV_EXCLUDE_FILE_TYPES",
-		"output":             "CFV_OUTPUT",
-		"reporter":           "CFV_REPORTER",
-		"groupby":            "CFV_GROUPBY",
-		"quiet":              "CFV_QUIET",
+	flag.Parse()
+
+	err := applyDefaultFlagsFromEnv()
+	if err != nil {
+		return validatorConfig{}, err
 	}
 
-	for flagName, envVar := range flagsEnvMap {
-		if err := setFlagFromEnvIfNotSet(flagName, envVar); err != nil {
-			return validatorConfig{}, err
-		}
+	reporterConf, err := parseReporterFlags(reporterConfigFlags)
+	if err != nil {
+		return validatorConfig{}, err
 	}
 
-	flag.Parse()
+	searchPaths := parseSearchPath()
 
-	searchPaths := make([]string, 0)
+	err = validateReporterConf(reporterConf, *groupOutputPtr)
```

Comment:
I would have expected this here


```suggestion
	err = validateReporterConf(reporterConf, groupOutputPtr)
```

The same way you pass the pointer to the other function